components/web_discovery/browser/regex_util.h

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_REGEX_UTIL_H_
+#define BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_REGEX_UTIL_H_
+
+#include <deque>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "base/containers/flat_map.h"
+#include "third_party/re2/src/re2/re2.h"
+
+namespace web_discovery {
+
+// Lazily creates and caches pre-compiled regexes, mainly used for
```

Comment:
are you sure it won't be accessible on different threads? Worth adding a comment that the class is not thread safe